### PR TITLE
front: use new editoast train_schedule endpoints

### LINKED ISSUE #4775: front: use new train_schedule endpoints
### Description and goal
Use the new editoast train_schedule endpoint to drop API

### Acceptance criteria
- [x] we use the new editoast train_schedule endpoint

### PR DESCRIPTION
**Pull request by clarani**
closes #4775

### CODE REVIEW

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx [hunk 1]
```diff
@@ -142,9 +142,10 @@ const Map: FC<MapProps> = ({ setExtViewport }) => {
     const concernedTrains: InterpoledTrain[] = [];
     simulation.trains.forEach((train, idx: number) => {
       const key = getRegimeKey(train.id);
```
Comment (anisometropie) [inline front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]: might as well do that everywhere :)
```suggestion
      const baseOrEco = getRegimeKey(train.id);
```

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx [hunk 2]
```diff
@@ -283,24 +284,29 @@ const Map: FC<MapProps> = ({ setExtViewport }) => {
...
-        dispatch(updateTimePositionValues(datetime2Isostring(timePositionLocal)));
-      } else {
-        throw new Error('Map onFeatureHover, try to update TimePositionValue with incorrect imput');
+      const train = selectedTrain[key];
+      if (train) {
+        const startCoordinates = getDirection(train.head_positions)
+          ? [geojsonPath.geometry.coordinates[0][0], geojsonPath.geometry.coordinates[0][1]]
+          : [
+              geojsonPath.geometry.coordinates[geojsonPath.geometry.coordinates.length - 1][0],
+              geojsonPath.geometry.coordinates[geojsonPath.geometry.coordinates.length - 1][1],
```
Comment (anisometropie) [inline front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]:
```suggestion
              last(geojsonPath.geometry.coordinates)[0],
              last(geojsonPath.geometry.coordinates)[1],
```

File: front/src/applications/stdcm/getStdcmTimetable.jsx [hunk 3]
```diff
@@ -27,19 +24,19 @@ export default async function getStdcmTimetable(
     if (!simulation.trains || !simulation.trains[selectedTrain]) {
       dispatch(updateSelectedTrainId(undefined));
     }
-    const timetable = await get(`${timetableURI}${timetableID}/`);
+    const timetable = await get(`/editoast/timetable/${timetableID}/`);
```
Comment (anisometropie) [inline front/src/applications/stdcm/getStdcmTimetable.jsx]: there is no editoast endpoint defined in rtk?

**Review by flomonster — APPROVED** [review]
Reviewed editoast part

File: front/src/applications/stdcm/views/StdcmRequestModal.jsx [hunk 4]
```diff
@@ -33,25 +29,35 @@ import { Spinner } from 'common/Loader';
...
+  const { t } = useTranslation([
+    'translation',
+    'operationalStudies/manageTrainSchedule',
+    'translation',
+  ]);
   const osrdconf = useSelector(getConf);
   const dispatch = useDispatch();
-  const [postStdcm] = osrdEditoastApi.usePostStdcmMutation();
+
+  const [postStdcm] = osrdEditoastApi.endpoints.postStdcm.useMutation();
```
Comment (anisometropie) [inline front/src/applications/stdcm/views/StdcmRequestModal.jsx:40]: why do you change this ?

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx (hunk 1 above)
Comment (clarani) [reply front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]: Done :)

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx (hunk 2 above)
Comment (clarani) [reply front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]: Done !

File: front/src/applications/stdcm/getStdcmTimetable.jsx (hunk 3 above)
Comment (clarani) [reply front/src/applications/stdcm/getStdcmTimetable.jsx]: As getStdcmTimetable is not a component, I can't use hooks here :/ That's why I only changed the url (and the params where needed), but we will have to refacto this later, when we will migrate this file to ts

File: front/src/applications/stdcm/views/StdcmRequestModal.jsx (hunk 4 above)
Comment (clarani) [reply front/src/applications/stdcm/views/StdcmRequestModal.jsx:40]: I find this clearer as you can directly access the endpoint in osrdEditoastApi by clicking on postStdcm, what do you think ?

File: front/src/applications/stdcm/getStdcmTimetable.jsx (hunk 3 above)
Comment (anisometropie) [reply front/src/applications/stdcm/getStdcmTimetable.jsx]: you can use rtk endpoints without hooks, but it’s messy.
But here you can just turn getStdcmTimetable into a custom hook to call hooks inside.

File: front/src/applications/stdcm/views/StdcmRequestModal.jsx (hunk 4 above)
Comment (anisometropie) [reply front/src/applications/stdcm/views/StdcmRequestModal.jsx:40]: that’s neat !
It even seems to provide better types infos. yeah keep this one

Screenshot_20230822_154653

File: front/src/applications/stdcm/getStdcmTimetable.jsx (hunk 3 above)
Comment (clarani) [reply front/src/applications/stdcm/getStdcmTimetable.jsx]: This file is not useful in the end, so I'm just deleting it :)

**Review by anisometropie — APPROVED** [review]

**Comment by flomonster** [comment]
It seems that some endpoint train schedules are still called on API. Is this intentional?

- PATCH: `/train_schedule/{id}/`
- GET: `/train_schedule/{id}/`
- POST: `/train_schedule/standalone_simulation/`

**Pull request merged**